### PR TITLE
Add clang_win-64

### PR DESCRIPTION
**Pull request by isuruf**
Checklist

- [ ] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [ ] Source is from official source
- [ ] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [ ] If static libraries are linked in, the license of the static library is packaged.
- [ ] Build number is 0
- [ ] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [ ] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

**Comment by isuruf** [comment]
@conda-forge/core, ready for review. Hope the license is okay as we are just downloading the proprietary stuff in an activate script in the user and not packaging it.

cc @mingwandroid

**Comment by mariusvniekerk** [comment]
So we may want to refer to https://aka.ms/WinSDKLicenseURL which is the canonical place to get the sdk license for windows?  Conveniently it is an rtf document.

**Comment by mingwandroid** [comment]
Great to see. The Windows SDK ISO is 796MB. Is there any way to request this ISO be cached by Azure? Does this matter?

**Comment by isuruf** [comment]
It doesn't matter on CI. A user has to set variable `CONDA_BUILD_WINSDK=/opt` and the sdk and msvc headers will be cached there.

File: .appveyor.yml [hunk 1]
```diff
@@ -4,7 +4,7 @@ skip_commits:
 environment:
 
   CONDA_INSTALL_LOCN: "C:\\Miniconda36-x64"
-  CONFIG: azure-win-64
+  CONFIG: win64
```
Comment (scopatz) [inline .appveyor.yml]: Are these all temporary changes?

Comment (isuruf) [reply .appveyor.yml]: No, I needed the $CI variable and these name changes were a mistake. They are actually used in https://github.com/conda-forge/staged-recipes/blob/master/.ci_support/build_all.py#L78 and somehow it was broken.

**Comment by isuruf** [comment]
@conda-forge/staged-recipes, ping again.

**Comment by beckermr** [comment]
@conda-forge/core you’d better look here

**Comment by isuruf** [comment]
ping

File: recipes/clang_win-64/LICENSE.txt [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+modification, are permitted provided that the following conditions are met:
+
+* Redistributions of source code must retain the above copyright notice, this
+  list of conditions and the following disclaimer.
+
+* Redistributions in binary form must reproduce the above copyright notice,
+  this list of conditions and the following disclaimer in the documentation
+  and/or other materials provided with the distribution.
+
+* Neither the name of staged-recipes nor the names of its
```
Comment (xhochy) [inline recipes/clang_win-64/LICENSE.txt]: Should this really be staged-recipes?
```suggestion
* Neither the name of clang_win-64 nor the names of its
```

File: recipes/clang_win-64/activate-msvc-headers.sh [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+MSVC_HEADERS_DIR=${CONDA_BUILD_WINSDK}/msvc-${MSVC_HEADERS_VERSION}
+
+if [[ ! -d "${MSVC_HEADERS_DIR}" ]]; then
+  mkdir -p "${MSVC_HEADERS_DIR}"
+  pushd "${MSVC_HEADERS_DIR}"
+    mkdir -p tmp
+    pushd tmp
+      curl -L -O https://www.nuget.org/api/v2/package/VisualCppTools.Community.VS2017Layout/${MSVC_HEADERS_VERSION}
+      unzip -o ${MSVC_HEADERS_VERSION} > /dev/null
+      #find lib -type d -name "x86" -delete || true
```
Comment (xhochy) [inline recipes/clang_win-64/activate-msvc-headers.sh]: What about those commented deletes? Will they be fixed or should they be removed?

Comment (isuruf) [reply recipes/clang_win-64/activate-msvc-headers.sh]: I was wondering whether we should keep non x64 libs around or not and decided to keep them. I kept the comment here as a reminder that we might want to do it later.

**Comment by isuruf** [comment]
> Should we keep a version number in the extraction folder so we can recognize whether it is consistent with the package version?

Yes, they go into `$CONDA_BUILD_WINSDK/msvc-14.11.25547` and `$CONDA_BUILD_WINSDK/winsdk-10.0.17134.0`

**Comment by xhochy** [comment]
> > Should we keep a version number in the extraction folder so we can recognize whether it is consistent with the package version?
> 
> Yes, they go into `$CONDA_BUILD_WINSDK/msvc-14.11.25547` and `$CONDA_BUILD_WINSDK/winsdk-10.0.17134.0`

That is the version number of the SDK, I was rather arguing for a version number of the extraction process that created it. Keeping this inside the conda env is probably unreasonable as `$CONDA_BUILD_WINSDK` will be reused between various environments. I'm ok with merging here without it but might be a thing to keep in mind once we have see that we missed extracting things, the winsdk version didn't change and we easily want to detect on activation whether the SDK needs fixing or not.

**Comment by isuruf** [comment]
Got it. I think we should validate that the `CONDA_BUILD_WINSDK` directory has a few files and error out if they are not instead of trying to repair it.

**Comment by chrisburr** [comment]
I’m also happy with it so LGTM :+1:

**Comment by isuruf** [comment]
Thanks for the reviews

**Pull request merged**